pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -476,6 +523,40 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		if err != nil {
 			return err
 		}
+
+		if useIOThreads {
+			ioThreadId := defaultIOThread
+			dedicatedThread := false
+			if disk.DedicatedIOThread != nil {
+				dedicatedThread = *disk.DedicatedIOThread
+			}
+
+			if dedicatedThread {
+				ioThreadId = currentDedicatedThread
+				currentDedicatedThread += 1
+
+				logger := log.DefaultLogger()
+				logger.Infof("Bus for this disk: %s", newDisk.Target.Bus)
+
+				if newDisk.Target.Bus == "scsi" {
+					newController := Controller{
+						Type:  "scsi",
+						Model: "virtio-scsi",
+						Driver: &ControllerDriver{
+							IOThread: &ioThreadId,
```

Comment:
Are we adding a controller for each scsi disk? IIRC, each virtio-scsi controller should support up to 256 disks.